lib/ansible/plugins/connection/ssh.py

```diff
@@ -506,6 +523,15 @@ def _build_command(self, binary, *other_args):
         # Next, additional arguments based on the configuration.
         #
 
+        # pkcs11 mode allows the use of Smartcards or Ubikey devices
+        if self._play_context.password and len(pkcs11_provider) >= 1:
```

Comment:
The resolution of the config/var switches discussion above should end up using the same thing here

Reply:
Thanks again.  I'll look for each place I check for this condition and update.